org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -431,19 +431,19 @@ private void configureDetectors(Map<String, Object> dmodels, ScanModel model, IP
 				.filter(IMalcolmModel.class::isInstance).map(IMalcolmModel.class::cast)
 				.findFirst().ifPresent(model -> req.getCompoundModel().setDuration(model.getExposureTime()));
 		}
-		
+
 		return service.createCompoundGenerator(req.getCompoundModel());
 	}
-	
+
 	private List<IRunnableDevice<?>> getDetectors(Map<String, ?> detectors) throws EventException {
-		
+
 		if (detectors==null) return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)